.github/workflows/generate-tidal-api.yml

```diff
@@ -0,0 +1,123 @@
+name: Generate API in the TidalAPI module and create a PR if there are changes
+
+on:
+  workflow_dispatch: # manual triggering of this workflow
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  codegen:
+    runs-on: macos-latest
+    env:
+      API_FOLDER: Sources/TidalAPI/
+      GENERATED_FOLDER_NAME: Generated
+      INPUT_FOLDER_NAME: Config/input
+      SCRIPT_NAME: generate_and_clean.sh
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Step 2: Install OpenAPI Generator via Homebrew
+      - name: Install OpenAPI Generator with Homebrew
+        run: |
+          brew update
+          brew install openapi-generator
+
+      - name: Run Code Generation Script
+        run: |
+          cd $API_FOLDER
+          chmod +x $SCRIPT_NAME
+          ./$SCRIPT_NAME
+
+      - name: Check for changes
+        id: check_for_changes
+        run: |
+          cd $API_FOLDER
+          git add . 
+          changes_in_generated=$(git status -s $GENERATED_FOLDER_NAME | sed 's/"//g')
```

Comment:
the combination of upper and lower snake case variables is a bit odd. Maybe settle on one style?

Reply:
The idea was to use `changes_in_generated` for local variables and `GENERATED_FOLDER_NAME` for output variables, to easier distinguish between them while reading the workflow file.

But I can for sure unify them, no problem!

Reply:
no, if that is the case I think it's all good! >Didn't see the system properly when glancing over it yesterday